external_parser/joiners/multistep_example_joiner.cc

```diff
@@ -280,7 +307,8 @@ bool multistep_example_joiner::process_joined(v_array<example *> &examples) {
     return false;
   }
 
-  joined.calc_reward(_loop_info.default_reward, _reward_calculation.value());
+  // TODO: remove
```

Comment:
is this to be removed in this PR? the access to the `joined.typed_data` pointer was introduced for binary to dsjson conversion, as that needs to know the ins and outs of the data but in general the pattern we followed was to delegate work to the `joined_event` and in turn to the typed data in `joined_event.h` instead of using that pointer in the joiner

Reply:
just removed.